crates/ide_assists/src/handlers/extract_struct_from_enum_variant.rs

```diff
@@ -230,14 +233,16 @@ fn update_reference(
 
     let module = ctx.sema.scope(&expr).module()?;
     if !visited_modules_set.contains(&module) {
-        if insert_import(ctx, rewriter, &expr, &module, enum_module_def, variant_hir_name).is_some()
-        {
+        if insert_import(ctx, &expr, &module, enum_module_def, variant_hir_name).is_some() {
             visited_modules_set.insert(module);
         }
     }
-    rewriter.insert_after(segment.syntax(), &make::token(T!['(']));
-    rewriter.insert_after(segment.syntax(), segment.syntax());
-    rewriter.insert_after(&expr, &make::token(T![')']));
+    ted::insert_raw(
+        ted::Position::before(segment.syntax()),
+        make::path_from_text(&format!("{}", segment)).clone_for_update().syntax(),
+    );
```

Comment:
I also realised that doing something like `insert(Position::after(some_node.syntax()), some_node.syntax())` either does nothing, or panic somewhere in `Vec::drain` in children splicing which is why I do this rather "fancy" deep clone here. Probably gonna disappear after I rewrite that assist but still good to bring up.

Reply:
Hmm, I thing we need a test for this on the rowan side.....

I thought about the case where you attach a node to its own descendant (creating a cycle), and that should be easy to catch. Attaching the node as an (indirect) sibling of itself feels like it might break some things....

Reply:
Oh ye I was wondering what would happen if one was to create an indirect cycle after I ran into this. I fear its not gonna end well...